cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/PipelineRegisterer.java

```diff
@@ -172,8 +172,12 @@ public Pipeline registerPlugins(ETLConfig config, Class errorDatasetType, Datase
     for (String stageName : stageTopologicalSortedOrder) {
       PipelineConfigureDetail pipelineConfigureDetail = stageToPipelineConfigureDetailMap.get(stageName);
       // configure pipeline in the topologically sorted order, to handle dependencies.
-      pipelineConfigureDetail.getPipelineConfigurable().configurePipeline(
-        pipelineConfigureDetail.getPipelineConfigurer());
+      try {
+        pipelineConfigureDetail.getPipelineConfigurable().configurePipeline(
+          pipelineConfigureDetail.getPipelineConfigurer());
+      } catch (IllegalArgumentException e) {
+        throw new IllegalArgumentException(String.format("Exception in stage : %s - %s", stageName, e.getMessage()), e);
```

Comment:
nit: think its a little clearer if we do "Exception in stage %s: %s". Can also pass in e's cause so that we don't have an illegal argument around an illegal argument.